src/model/bloodvesseljunction.hpp

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Stanford University, The Regents of the University of
+//               California, and others.
+//
+// All Rights Reserved.
+//
+// See Copyright-SimVascular.txt for additional details.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject
+// to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+// TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER
+// OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/**
+ * @file bloodvesseljunction.hpp
+ * @brief MODEL::BloodVesselJunction source file
+ */
+#ifndef SVZERODSOLVER_MODEL_BLOODVESSELJUNCTION_HPP_
+#define SVZERODSOLVER_MODEL_BLOODVESSELJUNCTION_HPP_
+
+#include "../algebra/sparsesystem.hpp"
+#include "block.hpp"
+#include "bloodvessel.hpp"
+
+namespace MODEL {
+/**
+ * @brief BloodVesselJunction
+ *
+ * Models a junction with one inlet and arbitrary outlets using
+ * MODEL::BloodVessel elements between each inlet and outlet pair.
+ *
+ * \f[
+ * \begin{circuitikz}
+ * \draw node[left] {$Q_{in}$} [-latex] (0,0) -- (0.8,0);
+ * \draw (1,0.1) node[anchor=south]{$P_{in}$};
+ * \draw (1,0) to [short, *-] (2.5,0.75);
+ * \draw (1,0) to [short, *-] (2.5,-0.75);
+ * \draw (2.5,0.75) node[anchor=south]{} to [generic, l_=$BV_{1}$, -*]
+ * (4.5,0.75); \draw (2.4,0.75) node[anchor=south]{$Q_{in,1}$}; \draw (4.6,0.75)
+ * node[anchor=south] {$P_{out,1}$}; \draw (2.5,-0.75) node[anchor=south]{} to
+ * [generic, l^=$BV_{2}$, -*] (4.5,-0.75); \draw (2.4,-0.75)
+ * node[anchor=north]{$Q_{in,2}$}; \draw (4.6,-0.75) node[anchor=north]
+ * {$P_{out,2}$}; \draw [-latex] (4.7,0.75) -- (5.5,0.75) node[right]
+ * {$Q_{out,1}$}; \draw [-latex] (4.7,-0.75) -- (5.5,-0.75) node[right]
+ * {$Q_{out,2}$}; \end{circuitikz} \f]
+ *
```

Comment:
Technical question: Can I view these somewhere while the pull request is still open?

Reply:
Yes, the html documentation is automatically built on GitHub. You can always go to the [last run of the Documentation routine](https://github.com/StanfordCBCL/svZeroDPlus/actions/runs/3525740588) and download the `documentation` artifact. You can then open `cpp`->`index.html` in the downloaded folder to view the html documentation locally.